### PR TITLE
added discount api and model class

### LINKED ISSUE #67: create discount-code model and api class
parent issue #66

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #67

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.fossasia.openevent.general.discount
+
+import io.reactivex.Single
+import org.fossasia.openevent.general.auth.DiscountCode
+import retrofit2.http.GET
+
+interface DiscountApi {
+
+    @GET("discount-codes?size=0")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: page[size]=0

File: app/src/main/java/org/fossasia/openevent/general/discount/DiscountCode.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonNaming
+import com.github.jasminb.jsonapi.IntegerIdHandler
+import com.github.jasminb.jsonapi.annotations.Id
+import com.github.jasminb.jsonapi.annotations.Type
+
+@Type("discount-code")
+@JsonNaming(PropertyNamingStrategy.KebabCaseStrategy::class)
+data class DiscountCode(
+        @Id(IntegerIdHandler::class)
+        val id: Int,
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/discount/DiscountCode.kt]: Add relationship with event

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountCode.kt]: okay

File: app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt (hunk 1 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: Also, discount codes should be loaded on per event basis

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: That means that there is only one discount code per event

About loading every discount code, in production code, there can be 1000s of events, not a good idea. Request will time out

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: the responses will be paged but, right? that means loading only fixed no of discount codes on every fetch

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: But then what'd be the point, the discount codes will be out of sync with the loaded events. You won't even be able to show which event discount code belongs to

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: got the point! what should be done then?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: Load the discount code when the event is clicked

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: okay!

**Comment by simarsingh24** [comment]
@iamareebjamal  please review

File: app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.fossasia.openevent.general.discount
+
+import io.reactivex.Single
+import org.fossasia.openevent.general.auth.DiscountCode
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+interface DiscountApi {
+
+    @GET("events/{event_identifier}/discount-code?include=event")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: Only include event ID, not the whole event

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: id, not event_identifier

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: okay cool!

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: you mean
``` @GET("events/{id}/discount-code?include=id")```

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: See orga app's API

**Comment by simarsingh24** [comment]
@iamareebjamal please review

**Pull request merged**